src/toil/cwl/cwltoil.py

```diff
@@ -691,25 +691,34 @@ def __init__(self, tool, cwljob, runtime_context, conditional):
     def run(self, file_store):
         cwljob = resolve_indirect(self.cwljob)
         fill_in_defaults(
-            self.cwltool.tool['inputs'], cwljob,
-            self.runtime_context.make_fs_access(
-                self.runtime_context.basedir or ""))
-        realjob = CWLJob(self.cwltool, cwljob, self.runtime_context, conditional=self.conditional)
+            self.cwltool.tool['inputs'],
+            cwljob,
+            self.runtime_context.make_fs_access(self.runtime_context.basedir or ""))
+        realjob = CWLJob(
+            tool=self.cwltool,
+            cwljob=cwljob,
+            runtime_context=self.runtime_context,
+            conditional=self.conditional)
         self.addChild(realjob)
         return realjob.rv()
 
 
 class CWLJob(Job):
     """Execute a CWL tool using cwltool.executors.SingleJobExecutor"""
+    def __init__(self,
+                 tool: ToilCommandLineTool,
+                 cwljob: dict,
+                 runtime_context: cwltool.context.RuntimeContext,
+                 step_inputs=None,
+                 conditional: Conditional = None):
```

Comment:
Did you run some kind of type checker and verify that all these annotations actually pass type checking?

Also here's another place where you might want `Optional` which is just a `Union` with `None`: https://stackoverflow.com/a/56670374

Reply:
Hmmm... it's not a bad idea to run mypy on just this file prior to running the tests.  Hopefully we can eventually convert the rest of the code base.

`Optional` has always seemed... cluttery?... to me.  It seems to be on most variables so it makes it less readable to me.  I like the explicit `Union[*, None]` though and will add those.